src/test/java/com/google/devtools/build/lib/skyframe/RecursiveFilesystemTraversalFunctionTest.java

```diff
@@ -671,9 +681,29 @@ private void assertTraversalOfDirectory(Artifact directoryArtifact) throws Excep
     appendToFile(someFile, "not all changes are treated equal");
     RecursiveFilesystemTraversalValue v4 =
         traverseAndAssertFiles(traversalRoot, expected1, expected2, expected3);
-    assertThat(v4).isEqualTo(v3);
+    assertThat(v4).isSameInstanceAs(v3);
     assertTraversalRootHashesAreEqual(v3, v4);
     assertThat(progressReceiver.invalidations).doesNotContain(traversalRoot);
+
+    // Add a new empty subdirectory to the directory and see that the value is rebuilt, but results
+    // in the collection of files.
+    // TODO(#15901): Empty directories currently aren't representable as tree artifact contents and
+    //  thus aren't tested here. Test this case when
+    //  https://github.com/bazelbuild/bazel/issues/15901 has been fixed.
```

Comment:
Last thing--let's drop the `Test this case when https://github.com/bazelbuild/bazel/issues/15901 has been fixed.`--that should happen as part of the checks anyway for any PR.

Reply:
I am also happy to delete that as part of the import--let me know how you would like to go about that.

Reply:
Yes, please delete on import. Thanks!

Reply:
Let me get to it then!